users/views.py

```diff
@@ -96,6 +104,22 @@ def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
     def get_success_url(self) -> str:
         return reverse("custom-users-list")
 
+    def form_valid(self, form: CustomUserCreationForm) -> HttpResponse:
+        if not self.request.is_ajax():
+            user_random_password = generate_random_string_from_letters_and_digits(
+                UserConstants.RANDOM_PASSWORD_LENGTH.value
+            )
+            user = form.save()
+            user.set_password(user_random_password)
+            user.is_active = True
+            user.need_change_password = True
+            user.full_clean()
+            user.save()
+            self._send_create_user_email(user, user_random_password)
+            logger.info(f"New user with id: {user.pk} has been created")
+            messages.success(self.request, ConfirmationMessages.SUCCESSFUL_CREATE_USER_MESSAGE)
```

Comment:
For sure it should be separate well named method

Reply:
Done.